src/Dialect/ONNX/ShapeInference/ONNXShapeHelper.cpp

```diff
@@ -65,6 +65,48 @@ ONNXOpShapeHelper<OP>::ONNXOpShapeHelper(OP *newOp, int numResults,
   };
 }
 
+template <>
+Value ONNXOpShapeHelper<Operation>::getOutput(int n) {
+  return op->getResult(n);
+}
+
+template <class OP>
+DimsExpr &ONNXOpShapeHelper<OP>::dimsForOutput(int n) {
+  Value output = getOutput(n);
+  // Try to combine the inferred shape and the output's shape if possbile.
+  if (isRankedShapedType(output.getType())) {
+    llvm::ArrayRef<int64_t> existingDims = getShape(output.getType());
+    // Do not handle the case of scalar tensor whose type can be tensor<f32>
+    // or tensor<1xf32>. Just use the inferredShape in this case.
+    if (existingDims.size() >= 1 && outputsDims[n].size() >= 1) {
```

Comment:
Why predicate all of this for sizes > 0? Size zero means no info?

Reply:
Size zero means `tensor<elementtype>` that is a scalar tensor. `tensor<1xelementype>` whose rank is 1 is also to represent a scalar tensor in onnx-mlir. They are used exchangebly in onnx-mlir. So not sure `tensor<elementtype>` or `tensor<1xelementtype>` is correct. It seems there is some op in yolov3 models that can handle both.

Reply:
got it, tx